backends/vulkan/CMakeLists.txt

```diff
@@ -92,6 +95,29 @@ target_link_libraries(vulkan_backend PRIVATE executorch)
 
 target_compile_options(vulkan_backend PRIVATE ${VULKAN_CXX_FLAGS})
 
+# This is required to ensure that vulkan_backend gets linked with
+# --whole-archive since backends are registered via static variables that would
+# otherwise be discarded
+target_link_options_shared_lib(vulkan_backend)
+
+# Executor Runner
+
+if(NOT CMAKE_TOOLCHAIN_FILE MATCHES ".*iOS\.cmake$")
+  set(VULKAN_RUNNER_SRCS ${_executor_runner__srcs})
+  list(TRANSFORM VULKAN_RUNNER_SRCS PREPEND "${EXECUTORCH_ROOT}/")
+  add_executable(vulkan_executor_runner ${VULKAN_RUNNER_SRCS})
+  target_link_libraries(vulkan_executor_runner ${_executor_runner_libs})
+  target_link_libraries(vulkan_executor_runner vulkan_schema)
+  target_link_libraries(vulkan_executor_runner vulkan_backend)
+  target_compile_options(vulkan_executor_runner PUBLIC ${VULKAN_CXX_FLAGS})
+
+  add_library(vulkan_executor_runner_lib STATIC ${VULKAN_RUNNER_SRCS})
+  target_link_libraries(vulkan_executor_runner_lib ${_executor_runner_libs})
+  target_link_libraries(vulkan_executor_runner_lib vulkan_schema)
```

Comment:
i remember there being some cmake.toml files you have to update as well?

Reply:
That makes it more convenient to collect the sources of a buck target, but it's not necessary for `executor_runner` targets because they don't define any new source files, they just add additional dependencies for specific backends. All `executor_runner` targets have pretty much the same source files, so I just used `_executor_runner__srcs` instead.

Another thing is I couldn't even update the `.toml` file because the Vulkan backend does not currently build with buck in open source. I had a PR to enable it, but I need to land some configerator changes to establish some OSS/Internal aliases so I haven't gotten around to it. I found that this works well and has the added bonus of not needing to introduce another `.toml` entry.